src/hw/raspi4/startup.s

```diff
@@ -0,0 +1,97 @@
+.section ".text.boot"
+
+.globl _Reset
+_Reset:
+  // read cpu id, stop slave cores
+  mrs x1, mpidr_el1
+  and x1, x1, #3
+  cbz x1, 2f
+  // cpu id > 0, stop
+1:wfe
+  b 1b
+2:// cpu id == 0
+  
+  // set stack before our code
+  ldr x1, =stack_top
+  
+  // set up EL1
+  mrs x0, CurrentEL
+  and x0, x0, #12 // clear reserved bits
+  
+  // running at EL3?
+  cmp x0, #12
+  bne 5f
+  // should never be executed, just for completeness
+  mov x2, #0x5b1
+  msr scr_el3, x2
+  mov x2, #0x3c9
+  msr spsr_el3, x2
+  adr x2, 5f
+  msr elr_el3, x2
+  eret
+  
+  // running at EL2?
+5:cmp x0, #4
```

Comment:
Please space this section out, I'd suggest a newline before each comment. Maybe 3/4 distinct blocks of code.